File: src/main/java/com/sah/game/piese/Cal.java
package com.sah.game.piese;

import com.sah.game.ChessBoard;
import com.sah.game.GameEnums.ColorType;
import com.sah.game.GameEnums.Tip;

public class Cal extends Piese {
    public Cal(ColorType color, int row, int col, ChessBoard game)
    {
        super(color, row, col, game);
        tip = Tip.KNIGHT;
    }

    @Override
    public boolean miscare(int targetRow, int targetCol)
    {
        if(patratValid(targetRow, targetCol) && peTabla(targetRow, targetCol))
            if(Math.abs(targetRow-row) * Math.abs(targetCol-col) == 2)
                return true;
        return false;
    }

    @Override
    public boolean poateAjunge(int targetRow, int targetCol)
    {
        if(Math.abs(targetRow-row) * Math.abs(targetCol-col) == 2)
            return true;
        return false;
    }
}

File: src/main/resources/static/ts/piese/Cal.ts
import {Piesa} from "./Piesa.js";
import {culoriPiesa, TipPiesa} from "../Enums.js";

export class Cal extends Piesa {
    readonly tip: TipPiesa = TipPiesa.KNIGHT;

    constructor(color: culoriPiesa, row: number, col: number) {
        super(color, row, col);
    }
}